src/workerd/api/form-data.c++

```diff
@@ -318,9 +318,10 @@ void FormData::parse(jsg::Lock& js,
               kj::Maybe<kj::StringPtr> maybeType, kj::ArrayPtr<const kj::byte> message) {
         KJ_IF_SOME(filename, maybeFilename) {
           if (convertFilesToStrings) {
+            auto messageData = kj::heapArray<char>(message.asChars());
             data.add(FormData::Entry{
-              .name = kj::str(name),
-              .value = kj::str(message.asChars()),
+              .name = js.accountedKjString(name),
+              .value = js.accountedKjString(kj::str(kj::mv(messageData))),
```

Comment:
Nit: I think the kj::str is redundant here, `accountedKjString(kj::Array<char>&& str)` can handle it directly right?

Reply:
Unfortunately no. The array is not null terminated. I'm going to look at it more later but passing it through kj::str was necessary to make it work.